packages/wrangler/src/index.ts

```diff
@@ -351,8 +433,13 @@ export function createCLIParser(argv: string[]) {
 		initHandler
 	);
 
-	// dev
-	register.registerNamespace("dev");
+	registry.define([
+		{
+			command: "wrangler dev",
+			definition: dev,
+		},
+	]);
+	registry.registerNamespace("dev");
```

Comment:
Can we merge `.define` and `.registerNamespace` into one api call?

Reply:
I don't believe we can. The limitation comes from the fact that Yargs requires all the registration of sub-commands up front. So you can't register `wrangler kv` and then later register `wrangler kv put`, because put needs to be registered in the same `.command` call to Yargs.

What this means is that first you need to `define` all the commands that you want, and _then_ you need to register them.

I suppose we could try to register all the commands that have been defined in the array immediately, by parsing the array for any new namespaces, but I think that could cause an issue if you wanted to define a command conditionally in a subsequent block and found that the namespace was already registered (and therefore you couldn't add to it).

Worth noting, however, that once everything is converted to the new API, we can drop the individual `registerNamespace` calls, and just keep one `registerAll` call at the end. The only reason we have to have `registerNamespace` at the moment is to ensure that everything is added to the Yargs object in the right order.

Reply:
Yeah, I agree with all of that. I'm wary about messing too much with the internals here though - ideally this PR just solves one problem, and we make further improvements in future PRs. What do you think?